src/codegen.cpp

```diff
@@ -533,6 +533,51 @@ static AttributeList get_attrs_zext(LLVMContext &C)
                 {Attributes(C, {Attribute::ZExt})});
 }
 
+static AttributeList get_attrs_ipoeffects(LLVMContext &C, jl_code_instance_t *ci)
+{
+    uint32_t e = ci->ipo_purity_bits;
+    uint8_t consistent = (e >> 0) & 0x07;
+    uint8_t effect_free = (e >> 3) & 0x03;
+    uint8_t nothrow = (e >> 5) & 0x01;
+    uint8_t terminates = (e >> 6) & 0x01;
+    uint8_t notaskstate = (e >> 7) & 0x01;
+    uint8_t inaccessiblememonly = (e >> 8) & 0x03;
+    uint8_t nonoverlayed = (e >> 10) & 0x01;
+#if JL_LLVM_VERSION >= 140000
+    AttrBuilder attr(C);
+#else
+    AttrBuilder attr;
+#endif
+    if (consistent == 0){
+        attr.addAttribute("consistent");
+    }
+    if (effect_free == 0)
+        attr.addAttribute("effect_free");
+    if (nothrow == 1){
+        attr.addAttribute("nothrow");
+        attr.addAttribute(Attribute::NoUnwind); // Is this even correct, because nothrow allows for try/catch
+    }
+    if (terminates == 1){
+        attr.addAttribute(Attribute::WillReturn);
+        attr.addAttribute("terminates");
+    }
+    if (consistent == 0 && terminates == 1 && effect_free == 0 && terminates == 1 && inaccessiblememonly == 0)
+        // attr.addAttribute(Attribute::Speculatable);
+    if (notaskstate == 1)
+        attr.addAttribute("notaskstate");
+    if (inaccessiblememonly == 0){
+        attr.addAttribute("inaccessiblememonly");
+        // attr.addAttribute(Attribute::InaccessibleMemOnly);
```

Comment:
I think it is probably okay to enable this attribute (if I remember correctly I made our semantics of `:inaccessiblememonly` effect bit same as LLVM's one).